### DISCUSSION TITLE
Timeline plugin custom values based on audio source

### DISCUSSION THREAD

**Discussion by Elgoritm** [katspaugh/wavesurfer.js#2822] [Q&A] [answered]
Hey there!

I am trying to set some custom values based on the audio file that's being loaded. My understanding so far is: 
I can create the timeline plugin instance, then create the wavesurfer instance, then load the audio. If I load the audio with a separate audio node, I can load it before, but by the time it figures out the values, it has already created the timeline plugin instance and I'm not getting custom values based on the audio.. 

Any help would be appreciated!

```
const topTimeline = TimelinePlugin.create({
  height: varA,
  insertPosition: 'beforebegin',
  timeInterval: varB,
  primaryLabelInterval: varC,
  secondaryLabelInterval: varD
  })
```

**Comment by katspaugh** [comment, marked as answer]
You can register plugins separately from wavesufer initialization by calling the `registerPlugin` function. E.g.

```js
const wavesufer = WaveSurfer.create({
  container: '#container',
  url: 'https://...',
})

wavesufer.on('ready', (duration) => {
  const timeline = TimelinePlugin.create({
     ...calculateVars(duration)
  })
  wavesufer.registerPlugin(timeline)
})
```

**Reply by Elgoritm** [reply]
Thank you for the quick reply, this worked! :)
